automation/jleechanorg_pr_automation/utils.py

```
@@ -0,0 +1,233 @@
+                    if os.path.exists(temp_path):
+                        try:
+                            os.remove(temp_path)
+                        except OSError:
+                            # Ignore errors during temp file cleanup (file may already be deleted)
+                            pass
+
+            except OSError as e:
+                logging.exception(f"Failed to write JSON to {file_path}: {e}")
+                return False
+
+    def update_json(self, file_path: str, update_func, lock_timeout: int = 10) -> bool:
```

Ruff (0.14.5)

Unused method argument: `lock_timeout`

(ARG002)